app/jobs/delete_host.rb

```diff
@@ -8,7 +8,13 @@ class DeleteHost
 
   def perform(message)
     host_id = message['id']
-    remove_related(host_id)
+    begin
+      num_removed = remove_related(host_id)
+    rescue StandardError => e
+      audit_fail(host_id, e)
+      raise
+    end
+    audit_success(host_id) if num_removed.positive?
```

Comment:
Probably our ruby version is mature enough for having an implicit catch block, at least if rubocop allows it here:
```ruby
  def perform(message)
    host_id = message['id']
    num_removed = remove_related(host_id)
  rescue StandardError => e
    audit_fail(host_id, e)
    raise
  else
    audit_success(host_id) if num_removed.positive?
  end
```
Maybe we have a global rule against this, let me know if that's the case :wink:

Reply:
I've used that concept in `handle_host_delete`. It could have been done also here...